File: routes/blog-routes.js
const express = require("express");
const { check } = require("express-validator");
const router = express.Router();
const blogControllers = require("../controllers/blog-controller");
const checkAuth = require("../middleware/check-auth");
router.get("/all", blogControllers.getAllBlogs);
router.get("/blog/:bid", blogControllers.getBlogById);
router.use(checkAuth);
router.post(
  "/comment/:bid",
  check("comment").notEmpty(),
  blogControllers.makeComment
);
router.patch(
  "/comment/:cid",
  check("comment").notEmpty(),
  blogControllers.updateComment
);
router.delete("/comment/:cid", blogControllers.deleteComment);
module.exports = router;

File: go_rewrite/routes/blog-routes.go
package routes

import (
	"backend/controllers"
	"backend/middleware"

	"github.com/gin-gonic/gin"
)

func BlogRoutes(router *gin.Engine) {
	router.GET("/blogs/all", controllers.GetAllBlogs)
	router.GET("/blogs/blog/:bid", controllers.GetBlogById)

	authorized:=router.Group("")

	authorized.Use(middleware.CheckAuth())
	authorized.POST("/blogs/comment/:bid", controllers.MakeComment)
	authorized.PATCH("/blogs/comment/:cid", controllers.UpdateComment)
	authorized.DELETE("/blogs/comment/:cid", controllers.DeleteComment)
}
